src/FakeItEasy/Core/TypeCatalogue.cs

```diff
@@ -91,7 +91,7 @@ private static IEnumerable<Assembly> GetAllAssemblies(IEnumerable<string> extraA
                     // Exclude the ReflectionOnly assemblies because we may fully load them later.
                     .Where(a => !a.ReflectionOnly)
 #endif
-                    .Where(a => !a.IsDynamic)
+                    .Where(a => !a.IsDynamic && !a.Name().Equals("DynamicProxyGenAssembly2", StringComparison.OrdinalIgnoreCase))

```

Comment:
I think I'm working toward understanding.

I think this is happening when DynamicProxyGenAssembly2 references FakeItEasy. Certainly, I can reproduce the problem in that situation, which occurs when tests that create Fakes have already been run and then we try to load the `TypeCatalogue`. In that case, I think that DynamicProxyGenAssembly2 is being stripped out by the `IsDynamic` check, but is added back in by

```c#
            return GetAssemblies(extraAssemblyFiles.Except(loadedAssemblyFiles))
-->             .Concat(loadedAssembliesReferencingFakeItEasy)                    // <---- this line here
                .Concat(FakeItEasyAssembly)
                .Distinct();
```

And that's when we have problems.

What think you?